kafka/consume.go

```diff
@@ -16,27 +16,32 @@ var ErrSeekTimedOut = errors.New("Kafka Seek timed out. Please try again.")
 
 // Consumer reads messages from Kafka.
 type Consumer interface {
-	io.Closer
-
 	// FetchMessage fetches one message, if there is any available at the current offset.
 	FetchMessage(ctx context.Context) (*kafkalib.Message, error)
 
+	// Close closes the consumer.
+	Close() error
+
 	// CommitMessage commits the offset of a given message.
 	CommitMessage(msg *kafkalib.Message) error
-}
 
-// OffsetAwareConsumer is a Consumer that can reset its offset.
-type OffsetAwareConsumer interface {
-	Consumer
+	// GetMetadata gets the metadata for a consumer.
+	GetMetadata(allTopics bool) (*kafkalib.Metadata, error)
 
-	// SetInitialOffset resets the current offset to the given one.
-	// Used for setting the initial offset a consumer should start consuming from.
-	// Should be called before start consuming messages.
-	SetInitialOffset(offset int64) error
+	// GetPartitions returns the partitions on the consumer.
+	GetPartions() ([]int32, error)
 
 	// Seek seeks the assigned topic partitions to the given offset.
-	// Seek() may only be used for partitions already being consumed.
-	Seek(offset int64, timeout time.Duration) error
+	Seek(offset int64) error
+
+	// SeekToTime seeks to the specified time.
+	SeekToTime(t time.Time) error
+
+	// SetPartittionByKey sets the current consumer to read from a partion by a hashed key.
+	SetPartitionByKey(key string, algorithm PartitionerAlgorithm) error
+
+	// SetPartitionByID sets the current consumer to read from the specified partition.
+	SetPartitionByID(id int32) error
```

Comment:
Can we skip this and move it out of the interface? I don't think it should be called from outside once the consumer is consuming.

Also a minor on the naming: I would prefer having more alignment in terms of naming with Kafka domain lang, so this could be named `AssignPartitionByID`, `AssignPartitionByKey` the other.

Reply:
This would simplify the code and those who depend on it

Reply:
As @phenixrizen explained to me via 📞 , this makes sense to stay there (perhaps just accepting my rename suggestion?) since this can be called once the consumer already started, making the consumer to switch to another partition without losing the TCP connection.